persistence/cosmosdb/index_outboxstorageformat_CosmosDB_[3.2,).partial.md

```diff
@@ -0,0 +1,9 @@
+A default [synthetic partition key](https://learn.microsoft.com/en-us/azure/cosmos-db/nosql/synthetic-partition-keys) will be used for all incoming messages, in the format `{endpointName}-{messageId}`, if not explicitly [overwritten](/persistence/cosmosdb/transactions.md#specifying-the-partitionkey-to-use-for-the-transaction) at runtime.
+
+> [!NOTE]
+> The [default partition key must be overwritten](/persistence/cosmosdb/transactions.md#specifying-the-partitionkey-to-use-for-the-transaction) whenever the message handler creates or updates business documents in CosmosDB. To guarantee atomicity, explicitly set the Outbox partition key to match the partition key of your business document (e.g., an `OrderId` or `CustomerId`). This is a requirement for including both the business document and the Outbox record in the same [Cosmos DB transactional batch](https://learn.microsoft.com/en-us/azure/cosmos-db/partitioning-overview). Conversely, for simplicity, you can use the default partition key when a handler's logic does not involve persisting business data.
+
+To support backward compatibility of control messages during migration, the persistence includes a fallback mechanism. When enabled (default), and if a record is not found using the synthetic key format, the system falls back to the legacy `{messageId}` format. The fallback can be disabled using `DisableReadFallback()` once all legacy records have expired, which is recommended since the fallback mechanism involves an additional read operation on the Outbox container.
```

Comment:
Perhaps put the  `DisableReadFallback()` into a code block that includes the object it is to be set on? So the sentence could become something like:

"Since the fallback mechanism involves an additional read operation on the Outbox container, it is recommended to turn it off once all legacy records have expired."

`config.DisableReadFallback()`